connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java

```diff
@@ -52,184 +53,274 @@ public class Loggers {
+            super(time);
+            loggerContext = (LoggerContext) LogManager.getContext(false);
+        }
+
+        @Override
+        public LoggerLevel level(String logger) {
+            Objects.requireNonNull(logger, "Logger may not be null");
+
+            org.apache.logging.log4j.Logger foundLogger = null;
+            if (isValidRootLoggerName(logger)) {
+                foundLogger = rootLogger();
+            } else {
+                var currentLoggers = currentLoggers().values();
+                // search within existing loggers for the given name.
+                // using LogManger.getLogger() will create a logger if it doesn't exist
+                // (potential leak since these don't get cleaned up).
+                for (org.apache.logging.log4j.Logger currentLogger : currentLoggers) {
+                    if (logger.equals(currentLogger.getName())) {
+                        foundLogger = currentLogger;
+                        break;
+                    }
+                }
             }
+
+            if (foundLogger == null) {
+                log.warn("Unable to find level for logger {}", logger);
+                return null;
+            }
+
+            return loggerLevel(foundLogger);
         }
-        Collections.sort(result);
 
-        return result;
-    }
+        @Override
+        public Map<String, LoggerLevel> allLevels() {
+            return currentLoggers()
+                    .values()
+                    .stream()
+                    .filter(logger -> !logger.getLevel().equals(Level.OFF))
+                    .collect(Collectors.toMap(
+                            this::getLoggerName,
+                            this::loggerLevel,
+                            (existing, replacing) -> replacing,
+                            TreeMap::new)
+                    );
+        }
 
-    /**
-     * Retrieve all known loggers within a given namespace, creating an ancestor logger for that
-     * namespace if one does not already exist
-     * @param namespace the namespace that the loggers should fall under; may not be null
-     * @return all loggers that fall under the given namespace; never null, and will always contain
-     * at least one logger (the ancestor logger for the namespace)
-     */
-    private synchronized Collection<org.apache.logging.log4j.Logger> loggers(String namespace) {
-        Objects.requireNonNull(namespace, "Logging namespace may not be null");
+        @Override
+        public List<String> setLevel(String namespace, String level) {
+            Objects.requireNonNull(namespace, "Logging namespace may not be null");
+            Objects.requireNonNull(level, "Level may not be null");
+            String internalNameSpace = isValidRootLoggerName(namespace) ? LogManager.ROOT_LOGGER_NAME : namespace;
+
+            log.info("Setting level of namespace {} and children to {}", internalNameSpace, level);
+
+            var loggers = loggers(internalNameSpace);
```

Comment:
In Kafka we're currently sticking to explicit types. It seems you only used `var` here so can you update them.

Reply:
I am not a fan of the `var` keyword either. I used it here, since it is present in the current code:

https://github.com/apache/kafka/blob/b92cd9d23647112b9a21fe0475605e0d5aca477a/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Loggers.java#L137-L139